autogpt_platform/backend/backend/integrations/webhooks/graph_lifecycle_hooks.py

```
@@ -198,15 +203,28 @@ async def on_graph_deactivate(graph: "GraphModel", user_id: str):
     for node in graph.nodes:
         block_input_schema = cast(BlockSchema, node.block.input_schema)
 
+        # First resolved credential wins. Assigning unconditionally per field
+        # meant a block with several credential fields passed only the last
+        # one, and a failed lookup on that last field discarded a credential
+        # an earlier field had resolved successfully.
         node_credentials = None
         for creds_field_name in block_input_schema.get_credentials_fields().keys():
```

Ruff (0.16.1)

[warning] Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)